plugins/guests/debian/cap/configure_networks.rb

```diff
@@ -11,6 +11,16 @@ class ConfigureNetworks
 
         def self.configure_networks(machine, networks)
           machine.communicate.tap do |comm|
+            main_interface = ''
+            comm.execute(%q!ip route | awk '$1=="default"{print $NF;exit}'!) do |type, data|
+              main_interface = data.chomp if type == :stdout
+            end
+            available_interfaces = []
+            comm.execute("ls /sys/class/net | grep -v -E '^(lo$|docker|lx[cd]|veth)'") do |type, data|
```

Comment:
I think commas in the regex `(e[n,t][h,s,p][[:digit:]]([a-z][[:digit:]])?` in https://github.com/mitchellh/vagrant/pull/6610/files should be deleted to say the least.

* [PredictableNetworkInterfaceNames](https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/)
* [systemd/udev-builtin-net_id.c at 25b0e6cb99c2a13c8f1ae9c5057f6941b722e764 · systemd/systemd](https://github.com/systemd/systemd/blob/25b0e6cb99c2a13c8f1ae9c5057f6941b722e764/src/udev/udev-builtin-net_id.c#L20)

I think `^(en|eth)` or `^e(n|th)` would do for the whitelist approach.
However [PredictableNetworkInterfaceNames](https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/) says
> You create your own manual naming scheme, for example by naming your interfaces "internet0", "dmz0" or "lan0"
So I think the blacklist approach may be better.

On the other hand, you can change the network interface name for lxd from the default `lxdbr0` to some other value. So the ideal way is detecting the type of network interfaces, not checking the names. But I had not found the way to do that, so I took the blacklist approach this time.